### PR TITLE
Add create/delete link capability

### PR DESCRIPTION
**Pull request by blazebissar**
Checkout the [Engineer Workflow Doc](https://docs.google.com/presentation/d/1pdzuHG-yO7_qW6jiByaGTLztKZ6b72lBNqSdNT5WZQ8/edit) to make sure you're contributing correctly! :)

**[Tickets](https://trello.com/b/dYOQKJGv/reclass)** 
Trello Ticket:
Additional Notes:
Link creation/deletion capability in the topology modification series of PRs. This one is the most complex of any of these because of all the logic needed to ensure links are correct. I think the test cases are fairly robust but might have missed some edge cases.

**[Dependencies](https://docs.google.com/document/d/1CM6OA2nlhI3DHnHdDEgySukx_tZ_wsG-C2YgraZRvJA/edit#heading=h.czpk4cbi7z4g)**
None.
**[Explain Impact on DB](https://docs.google.com/document/d/1CM6OA2nlhI3DHnHdDEgySukx_tZ_wsG-C2YgraZRvJA/edit#heading=h.285lgcs4z84o)**
None.
**[Testing](https://docs.google.com/document/d/1CM6OA2nlhI3DHnHdDEgySukx_tZ_wsG-C2YgraZRvJA/edit#heading=h.rs692kr58cf8)**

How did you test this? List of actions & expected behavior with helpful screenshots for reviewer:
New unit tests written for creation/deletion in separate function.
If this change is performance related, please document what tests were / should be done to verify improvement:
None.

### CODE REVIEW

File: flasksrc/__init__.py [hunk 1]
```diff
@@ -80,6 +81,10 @@ def create_app(test_config=None):
                      '/api/toynet/session/<int:toynet_session_id>/create/switch')
     api.add_resource(ToyNetSessionByIdCreateRouter,
                      '/api/toynet/session/<int:toynet_session_id>/create/router')
+    api.add_resource(ToyNetSessionByIdCreateLink,
+                     '/api/toynet/session/<int:toynet_session_id>/create/link')
+    api.add_resource(ToyNetSessionByIdDeleteLink,
+                     '/api/toynet/session/<int:toynet_session_id>/delete/link')
```
Comment (scottjr632) [inline flasksrc/__init__.py:87]: Why not just make one endpoint `/api/toynet/session/<int:toynet_session_id>/link` and have it handle multiple HTTP methods, (i.e. DELETE, POST)?

File: flasksrc/session.py [hunk 2]
```diff
@@ -556,6 +575,234 @@ def put(self, toynet_session_id, **kwargs):
...
+
+        # Find the specified devices
+        dev_list = root.find('routerList').findall('router')
+        dev_list += root.find('switchList').findall('switch')
+        dev_list += root.find('hostList').findall('host')
+        for device in dev_list:
+            if device.get('name') == req['dev_1']:
+                dev1_elem = device
+            if device.get('name') == req['dev_2']:
+                dev2_elem = device
```
Comment (scottjr632) [inline flasksrc/session.py:750]: We use `from` and `to` to indicate a link in the frontend, not sure how you feel about that nomenclature though.

```suggestion
            if device.get('name') == req['from']:
                dev1_elem = device
            if device.get('name') == req['to']:
                dev2_elem = device
```

**Review by scottjr632 — COMMENTED** [review]
Just some general comments. I have not gotten to run this with the frontend yet but will do that on Monday.

File: flasksrc/__init__.py (hunk 1 above)
Comment (blazebissar) [reply flasksrc/__init__.py:87]: I wanted to keep the same convention as the devices where it is a `/[create|delete]/[name]` endpoint for consistency.

File: flasksrc/session.py (hunk 2 above)
Comment (blazebissar) [reply flasksrc/session.py:750]: I chose this nomenclature since links are undirected; if `to` and `from` are prevalent in the frontend I'm open to changing them here to keep things simpler though.

File: flasksrc/session.py [hunk 3]
```diff
@@ -77,6 +78,15 @@ class ToyNetSessionByIdCreateSwitchPutResp(Schema):
     pass
 
 
+class ToyNetSessionByIdCreateLinkPutReq(Schema):
+    dev_1 = fields.Str()
+    dev_2 = fields.Str()
+
+
+class ToyNetSessionByIdCreateLinkPutResp(Schema):
+    pass
```
Comment (ajlacey) [inline flasksrc/session.py:87]: What is the purpose for not implementing this? Is there an expected future use?

Comment (blazebissar) [reply flasksrc/session.py:87]: Partially to enable easy modification should the API change, partially to ensure that Swagger generates complete documentation (even containing empty entries), and partially to guard against errors of omission or perceived errors of omission.
